file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+        200:
+          description: OK
+          schema:
+            description: Version information
+            title: RootEndpointResponse
+            type: object
+            properties:
+              v1:
+                $ref: '#/definitions/V1Operations'
+              version:
+                description: The implementation version of the web service
+                type: string
+  /{version}:
+    parameters:
+      - in: path
+        name: version
+        description: The version of the API to retrieve operations.
+        type: string
+        required: true
+    get:
+      tags: [versioning]
+      summary: API version information
+      description: Returns available operations for a single version of the API.
+      operationId: RootEndpointWithVersion
+      x-ni-request-all-privileges: true
+      # Explicitly mark security as an empty array - this route does not require any privileges.
+      # Marking it this way prevents it from inheriting the top-level security settings.
+      security: []
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/V1Operations'
+        404:
+          description: Not Found
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/service-groups:
+    get:
+      tags: [service groups]
+      summary: List service groups
+      description: >-
+        Returns an array of File service group names. Each service group is a
+        distinct collection of files.
+      operationId: ListServiceGroups
+      x-ni-operation: listFiles
+      x-ni-privilege: ViewMetadata
+      x-ni-request-all-privileges: true
+      responses:
+        200:
+          description: OK
+          schema:
+            description: Service group information
+            title: ListServiceGroupsResponse
+            type: object
+            properties:
+              _links:
+                type: object
+                description: >-
+                  Links that apply to the collection of service groups:
+
+                  - self: Link to the collection of service groups
+                properties:
+                  self:
+                    $ref: '#/definitions/Link'
+              serviceGroups:
+                type: array
+                description: Array of service groups within the file service
+                items:
+                  $ref: '#/definitions/ServiceGroup'
+        401:
+          $ref: '#/responses/Unauthorized'
+  /v1/service-groups/Default/files:
+    get:
+      tags: [files]
+      summary: List available files
+      description: >-
+        Lists available files on the Skyline File service.
+
+        The skip and take parameters can be used return paged responses. The
+        user can skip an unsigned integer number of entries, and take an
+        unsigned integer number of entries. A list of 100 files with a skip
+        value of 50 and a take value of 25 will return entries 51 through 75.
```

Comment:
Use the skip and take parameters together to return paged responses. Specify a number of entries to skip and a number of entries to return after skipping. For example, a list of 100 files with a skip value of 50 and a take value of 25 will return entries 51 through 75.

Reply:
I'm also wondering if any of this belongs in parameter descriptions instead of the route description. Just something to think about.